touchforms/backend/xformplayer.py

```diff
@@ -322,11 +322,9 @@ def ix_in_scope(form_ix):
             evt = self.__parse_event(form_ix)
             evt['relevant'] = relevant
             if evt['type'] == 'sub-group':
-                presentation_group = (evt['caption'] != None)
-                if presentation_group:
-                    siblings.append(evt)
-                    evt['children'] = []
-                form_ix = self._walk(form_ix, evt['children'] if presentation_group else siblings)
+                siblings.append(evt)
+                evt['children'] = []
+                form_ix = self._walk(form_ix, evt['children'])
```

Comment:
so my impression is that when you call it with "siblings" it changes the structure of the return object to to be a flat list:
 
with presentation group:
```
question1
repeat
  question2
  question3
```

without a presentation group it changes it to a flat list:
```
question1
question2
question3
```
but this breaks TF because the indexes are now off. i have no idea if this was ever working. it's possible that i broke it a long time ago when i refactored TF frontend and it was no longer able to handle that case. groups without labels were probably just extremely rare. either way it seems silly to keep around